src/uproot/interpretation/objects.py

```diff
@@ -291,11 +312,15 @@ def _discover_forth(self, data, byte_offsets, branch, context, cursor_offset):
                 branch.file.detached,
                 branch,
             )
+
             if context["cancel_forth"] and "forth" in context.keys():
                 del context["forth"]
+
             if "forth" in context.keys():
                 context["forth"].gen.awkward_model = context["forth"].gen.top_node
-                if not context["forth"].gen.var_set:
+
+                discovered_form = context["forth"].gen.top_form
+                if not self._any_NULL(discovered_form):
```

Comment:
This is the essence of the change. We no longer look at `var_set` (or even construct it), but instead ask if there are any `"NULL"` values within the Form.